examples/async_await/bin/async_example.dart

```diff
@@ -12,7 +12,7 @@ Future<String> fetchUserOrder() {
   return Future.delayed(const Duration(seconds: 4), () => 'Large Latte');
 }
 
-Future<void> main() async {
+void main() async {
```

Comment:
This should trigger a lint warning. An `async` function always returns a `Future`.

Reply:
That's the [`avoid_void_async`](https://dart.dev/tools/linter-rules#avoid_void_async) rule which we don't have enabled.

I am missing some context but it seems the language team decided it's best to discourage `Future<void>` for the `main` function or at least not showcase it in examples. Perhaps because it suggests the result is useful. There is [a linter issue](https://github.com/dart-lang/linter/issues/3141) considering adding it as an exception to the lint.

Hoping @mit-mit can provide some more thoughts.

Reply:
Oh, that's new to me. If @mit-mit is happy with this PR, I'm happy.

Reply:
You are right generally @domesticmouse but we discussed making an exception for `main` specifically. @pq @lrhn should we adjust the lint to include that exception?

Reply:
I'd be happy with changing the lint to not trigger on `main` functions. 
There is no official rules to not have `void` `async` functions, the lint is not part of the Dart team's official `lints` package, probably because there is too high a risk of false positives (there are reasonable cases for having a `void`/`async` function, they are just not easily distinguishable from doing it by mistake). 
Not triggering on `main` might reduce the false positives.

(Personally, I'll likely never enable the lint, because I am likely to get false positives. I write the kind of code that gets those false positives. I know Google's internal code style is stricter about disallowing things that can *potentially* be an error, and that's fine too. That's why it's a lint, but not a recommended one, so people can opt-in if they want it.)

Reply:
I'm in favor of special-casing `main` in `avoid_void_async ` and have updated https://github.com/dart-lang/linter/issues/3141 to a "proposal".  Unless there are objections, I'll go for it.